lib/ripper-tags/json_formatter.rb

```diff
@@ -1,10 +1,37 @@
-require 'yajl'
+begin
+  require 'yajl/json_gem' unless defined?(::JSON)
+rescue LoadError
+  require 'json'
+end
 require 'ripper-tags/default_formatter'
 
 module RipperTags
   class JSONFormatter < DefaultFormatter
-    def format(tag)
-      Yajl.dump(tag)
+    def supported_flags() ['s'] end
+
+    def stream_format?
+      return @stream_format if defined? @stream_format
+      @stream_format = extra_flag?('s')
+    end
+
+    def with_output
+      super do |true_out|
+        if stream_format?
+          yield true_out
+        else
+          buffer = []
+          yield buffer
+          true_out << ::JSON.dump(buffer)
+        end
+      end
+    end
+
+    def write(tag, buffer)
+      if stream_format?
+        buffer.puts JSON.dump(tag)
```

Comment:
I've added back the original stream format under the `--extra=s` option ("s" is for "stream"). This still keeps the giant JSON array format as default. Let me know if you want it to be switched around.